### PR TITLE
V2 collection create2 deploy

### PR DESCRIPTION
**Pull request by karmacoma-eth**

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ force = false
 # The verbosity of tests
 verbosity = 3
 
-solc_version = '0.8.17'
+solc_version = '0.8.16'
```
Comment (jorgeavaldez) [inline foundry.toml:17]: :scream:

File: script/utils/Create2Helper.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.0 <0.9.0;
+
+import {StdUtils} from "forge-std/StdUtils.sol";
+
+abstract contract Create2Helper is StdUtils {
```
Comment (jorgeavaldez) [inline script/utils/Create2Helper.sol:6]: :fire:

File: src/ShowtimeVerifier.sol [hunk 3]
```diff
@@ -23,7 +23,7 @@ contract ShowtimeVerifier is Ownable, EIP712, IShowtimeVerifier {
                             MUTABLE STORAGE
     //////////////////////////////////////////////////////////////*/
 
-    mapping(address => uint256) public signerValidity;
+    mapping(address => uint256) public  ;
```
Comment (jorgeavaldez) [inline src/ShowtimeVerifier.sol]: is this ok?

File: script/GatedEditions.s.sol [hunk 4]
```diff
@@ -2,31 +2,67 @@
...
+    bytes32 internal creatorSalt =
+        bytes32(uint256(49878374740187518875513948233179854588023069277032562717373590206714954727220));
+
+    function _log(string memory s, bytes32 v) public view {
+        console2.log(s);
+        console2.logBytes32(v);
+    }
+
+    function _checkPrefix(address addr) public {
+        assertEq(uint256(uint160(addr)) >> 136, 0x50c001);
```
Comment (jorgeavaldez) [inline script/GatedEditions.s.sol:31]: :heart:

File: bin/backend-sign.py [hunk 5]
```diff
@@ -41,23 +41,11 @@ def create_keyfile():
 def get_verifier_abi():
     return [
         {
-            "inputs": [
```
Comment (jorgeavaldez) [inline bin/backend-sign.py:44]: shoutout to the formatter

**Review by jorgeavaldez — APPROVED** [review]

File: src/ShowtimeVerifier.sol (hunk 3 above)
Comment (jorgeavaldez) [reply src/ShowtimeVerifier.sol]: i see it was fixed before i submitted hahaha

File: foundry.toml (hunk 1 above)
Comment (karmacoma-eth) [reply foundry.toml:17]: I know :(

File: bin/backend-sign.py (hunk 5 above)
Comment (karmacoma-eth) [reply bin/backend-sign.py:44]: 🙈

File: src/ShowtimeVerifier.sol (hunk 3 above)
Comment (karmacoma-eth) [reply src/ShowtimeVerifier.sol]: no, t'was a bad commit that I rebased out of existence

**Pull request merged**